File: elixir/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    mp = %{
      black: 0,
      brown: 1,
      red: 2,
      orange: 3,
      yellow: 4,
      green: 5,
      blue: 6,
      violet: 7,
      grey: 8,
      white: 9
    }

    colors
    |> Enum.take(2)
    |> Enum.reduce(0, fn color, acc ->
      acc * 10 + mp[color]
    end)
  end
end

File: dart/resistor_color_duo.dart
class ResistorColorDuo {
  List<String> get colors {
    return [
      'black',
      'brown',
      'red',
      'orange',
      'yellow',
      'green',
      'blue',
      'violet',
      'grey',
      'white'
    ];
  }

  int colorCode(String color) {
    return colors.indexOf(color);
  }

  int value(List<String> colors) {
    return colors.sublist(0, 2).fold(0, (acc, c) => acc * 10 + colorCode(c));
  }
}
